### ISSUE TITLE
[Feature]: Custom fields support

### ISSUE THREAD

**Issue by QazCetelic** [astubenbord/paperless-mobile#420]
### Description

The Paperless web interface has a feature to add custom fields to documents:

<details>
<summary>Screenshot</summary>

![Screenshot](https://github.com/astubenbord/paperless-mobile/assets/51381523/bdc81a51-38bb-4ef7-a6a9-62b222e5564e)

</details>

These are then visible when viewing the document:

<details>
<summary>Screenshot</summary>

![Screenshot](https://github.com/astubenbord/paperless-mobile/assets/51381523/c71e41a1-11bc-4464-8824-959301a96324)

</details>

It would be nice to also be able to view and edit these with this app.

I could take a look to see if I can implement it myself, but I have very limited experience with Flutter. I did try to clone the project, but I got 13525 compile errors which I don't know how to solve.

**Comment by astubenbord** [comment]
Hi, it already works in my current development state. However, I have quite a lot of changes there, so it might still take a while until I can publish a new update. Unfortunately I don't have that much spare time to work on this project at the moment, but there's light at the end of the tunnel, so hopefully I'll get to work on this more frequently in the upcoming weeks!

**Comment by Krabtus** [comment]
Is there any news yet about whether it will soon be possible to manage custom fields via the mobile app?

**Comment by royaltongue** [comment]
Would also like to see this! Custom fields are great for linking certain documents of different types or tags together

**Comment by nagarrido** [comment]
Would this include filters?

**Comment by freeDom-** [comment]
I am also very much interested in this feature.
Could you not split your releases into small sub packages? This helps a lot to not lose the track in bigger projects and gwt changes done.

**Comment by Tibabern** [comment]
I am also interested in a solution because all my documents  have an additional field and because of it I can't edit them in the app. Many thanks.

**Comment by hyperinfiltrat** [comment]
Are there any news on the topic of implementing "custom fields"?
Kind regards and many thanks!

**Comment by p18dh28ejfi** [comment]
Any updates on this? This makes the android app for me unusable, as 99% of my documents have a custom field.
Thanks in advance!

**Comment by Krabtus** [comment]
Unfortunately, it seems that the Android app Paperless Mobile is no longer being developed. However, an Android app called **Papernext** was recently introduced, and it looks very good.

**Comment by astubenbord** [comment]
Please check out the newest pre-release (available on [GitHub](https://github.com/astubenbord/paperless-mobile/releases/tag/v4.1.0) and in Google Play Beta Program in ~30 minutes after the build finishes). Please let me know if you encounter any bugs or unexpected behavior.

**Comment by Limerick-gh** [comment]
Hi, just tested latest release and I can show and edit custom fields which have been assigned  by a workflow. 😃👍🏻
What I currently cannot get to work on my end is to delete a custom field. Clicking the bin icon removes it, but it is coming back when saving the document.

**Comment by astubenbord** [comment]
@Limerick-gh appreciate the feedback, I'll have a look.

**Comment by astubenbord** [comment]
@Limerick-gh how many custom fields do you have on your documents? I just found a possibly related bug: When form fields are scrolled out of view, the elements get recycled and therefore their form value is reset to null. I've had the issue that with many custom fields, scrolling all the way down resets title etc. causing saving changes to fail. So if you have lots of custom fields, or a small device, scrolling all the way back to the top could reset the custom field form causing changes to get lost. Could that be the issue? I'll upload a patched version.

**Comment by Limerick-gh** [comment]
Probably the issue is on my end. 😆
I used a tag to add custom fields by using a workflow.

On paperless website, I can delete custom fields while keeping the tag assigned. They remain removed when saving the doc.
On paperless mobile app, it seems to re-add the tags once saved, when the workflow tag remains on the doc. If I remove the tag, custom fields deletion works as expected.
Could it be that workflows are refrigerated again?
I have to admit that trigger is 'document updated'. 
So probably it's more that the behavior of the pngx website  is not correct. 😉

**Comment by astubenbord** [comment]
Hm, I haven't used workflows yet, so I can't say much about it. How did you setup your workflow? Trigger is document update, and action is ...? I'll try to reproduce the issue once I know how your workflow is configured :)

**Comment by Limerick-gh** [comment]
It's basically:

- Trigger: Document updated
- Advanced Filter: Has a tag 'invoice'

- Action Type: Assignment
- Assign Custom fields: Date field, true/false field

But again, I'd say the behavior of your app is actually correct. I was just used to the website behavior, which is different, but cannot say why fields are not re-added if document is saved. While thinking about it, not re-addding them is actually more unexpected, but it it helped me in the way I used custom fields.

Background is that I like to have additional fields for invoices, like Payment Date and a toggle for Paid status etc. So I add them by the workflow. But I don't need all of them at every time. So in some docs I remove the custom fields.

**Comment by astubenbord** [comment]
I see, so I just recreated the workflow and removing the custom field indeed has no effect in the web client if the tag stays assigned to the document. Maybe it's me, but I don't see a difference in the behavior of the app and web client in the way you described. When I remove the custom field from the document, save changes, close the document in the web UI and re-open it, the custom field is re-assigned again. But yeah, kind of expected but also unexpected behavior, no matter how it's implemented :D

**Comment by Limerick-gh** [comment]
You're right, my bad. Guess I was tricked by the browser cache, while jumping forth and back. 🫣
Thanks for excellent app! 💯
